joomla/joomla-cms
# [4.0] highlight js

Pull Request for Issue #32997 redo of #16357 .

### Summary of Changes
Use abetter script for marking HTML: https://markjs.io https://bundlephobia.com/result?p=mark.js@8.11.1


### Testing Instructions

After pulling this PR and running `npm ci`:

Create an article with the word "create" and the word "admin" in the body text.
Ensure the System - Highlight plugin is enabled
View the article on the frontend
Then append the url you are on with ?highlight=WyJjcmVhdGUiLCJhZG1pbiJd

WyJjcmVhdGUiLCJhZG1pbiJd = base64encode of the array ["create","admin"]


### Actual result BEFORE applying this Pull Request
No highlight


### Expected result AFTER applying this Pull Request
![Screenshot 2021-04-10 at 15 08 19](https://user-images.githubusercontent.com/3889375/114270802-a269e280-9a0e-11eb-8d86-143b16105b3f.png)



### Documentation Changes Required

This should be 100% B/C but the idea is to deprecate as many HTMLHelpers as possible as they are really unfriendly for front end devs (nobody will ever create a Plugin just to adjust some HTML/JS/CSS and then have to maintain it)


@PhilETaylor can you give this a test?
